crates/frontend/src/pages/snapshot_list.rs

```diff
@@ -0,0 +1,216 @@
+            |(current_tenant, page, count)| async move {
+                match fetch_snapshots(current_tenant.to_string(), page, count).await {
+                    Ok(response) => response.data,
+                    Err(_) => vec![],
+                }
+            },
+        );
+
+    let handle_next_click = Callback::new(move |total_pages: i64| {
+        set_filters.update(|f| {
+            f.page = match f.page {
+                Some(p) if p < total_pages => Some(p + 1),
+                Some(p) => Some(p),
+                None => Some(1),
+            }
+        });
+    });
+
+    let handle_prev_click = Callback::new(move |_| {
+        set_filters.update(|f| {
+            f.page = match f.page {
+                Some(p) if p > 1 => Some(p - 1),
+                Some(_) => Some(1),
+                None => Some(1),
+            }
+        });
+    });
+
+    view! {
+        <div class="p-8">
+            <Suspense fallback=move || view! { <Skeleton/> }>
+                <div class="pb-4">
+                    {move || {
+                        let snapshot_res = snapshots_resource.get();
+                        let total_items = match snapshot_res {
+                            Some(snapshot_resp) => snapshot_resp.len().to_string(),
+                            _ => "0".to_string(),
+                        };
+                        view! {
+                            <Stat
+                                heading="ConfigVersions"
+                                icon="ri-camera-lens-fill"
+                                number=total_items
+                            />
+                        }
+                    }}
+
+                </div>
+                <div class="card rounded-xl w-full bg-base-100 shadow">
+                    <div class="card-body">
+                        <div class="flex justify-between">
+                            <h2 class="card-title">Config Versions</h2>
+                        </div>
+                        <div>
+                            {move || {
+                                let value = snapshots_resource.get();
+                                let filters = filters.get();
+                                match value {
+                                    Some(v) => {
+                                        let page = filters.page.unwrap_or(1);
+                                        let count = filters.count.unwrap_or(10);
+                                        let total_items = v.len();
+                                        let total_pages = (total_items as f64 / count as f64).ceil()
+                                            as i64;
+                                        let start = ((page - 1) * count as i64) as usize;
+                                        let end = std::cmp::min(
+                                            start + count as usize,
+                                            total_items,
+                                        );
```

Comment:
I don't think this logic is needed, anyway we just have current page data, so this calculation is redundant.